test/integration/test_metadata.py

```diff
@@ -0,0 +1,54 @@
+import os
+import json
+from time import sleep
+
+import pytest as pytest
+
+from FLIR.conservator.generated.schema import AnnotationCreate
+
+# Unfortunately, metadata references specific IDs, so
+# can't be hard-coded test data. Also, can't use a class
+# and split download/upload because file system is wiped
+# between tests.
+
+
+def test_metadata_download_upload_for_media(conservator, test_data, tmp_cwd):
+    collection = conservator.collections.create_from_remote_path("/Some/Collection")
+    local_path = test_data / "jpg" / "bicycle_0.jpg"
+    media_id = conservator.media.upload(local_path, collection=collection)
+    conservator.media.wait_for_processing(media_id)
+
+    image = conservator.images.all().first()
+    frame = image.get_frame()
+    annotation = AnnotationCreate(
+        labels=["abc", "def"], bounding_box={"x": 1, "y": 2, "w": 3, "h": 4}
+    )
+    frame.add_annotation(annotation)
+
+    # First download
+    image.download_metadata(".")
+
+    assert os.path.exists("bicycle_0.json")
+    assert os.path.isfile("bicycle_0.json")
+    with open("bicycle_0.json") as f:
+        local_metadata = json.load(f)
+    assert local_metadata == json.loads(image.metadata)
+
+    # Then, we modify local metadata
+    annotation = local_metadata["videos"][0]["frames"][0]["annotations"][0]
```

Comment:
It might be worthwhile to double-check the metadata format here.  There was a time when the certain arrays would be empty or even missing if there were no annotations or custom frame metadata associated with the media file.  So e.g., check specifically for "videos" in local_metadata (and not empty), "frames" in local_metadata["videos"] (and not empty), etc.  I'm not sure, but it may be more informative than the key error or indexing into an empty array exception you'd otherwise get.

Reply:
At the least, those fields would be checked before trying to access them.

Reply:
Sure, I will add some checks for the fields before accessing them